src/duk_error_throw.c

```diff
@@ -107,10 +113,11 @@ DUK_INTERNAL void duk_err_create_and_throw(duk_hthread *thr, duk_errcode_t code)
 	 *  Finally, longjmp
 	 */
 
-	thr->heap->handling_error = 0;
-
 	duk_err_setup_heap_ljstate(thr, DUK_LJ_TYPE_THROW);
 
+	thr->callstack_max -= DUK_CALLSTACK_DEFAULT_MAX * 2;  /* reset reclimit */
+	thr->heap->handling_error = 0;
```

Comment:
Asserting and setting back to DUK_CALLSTACK_DEFAULT_MAX might be the best. It's good for the assert to fail early rather than possibly quite a long time before the next error occurs. Setting it back is probably better than staying off-sync when asserts are disabled.